driver-core/src/main/com/mongodb/internal/logging/StructuredLoggingInterceptor.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.internal.logging;
+
+/**
+ * <p>This class is not part of the public API and may be removed or changed at any time</p>
+ */
+public interface StructuredLoggingInterceptor {
```

Comment:
Shouldn't this interface be `@ThreadSafe`, which would also mean that `TestLoggingInterceptor` needs to be implemented in a thread-safe manner?

Reply:
There is no good way to use the test interceptor in a scenario where there are multiple threads, since it relies on the order of the log messages.  If we ever do get into those types of scenarios, we can address it then.  

I suspect that most of our test infrastructure code is not actually thread safe for this or similar reasons.

Reply:
Done (and also annotated with `FunctionalInterface`.  I consider just replacing with `Consumer<StructuredLogMessage` but left it alone.